File: FinalCode/brownianMotion.py
import math
import numpy as np

def brownianMotion(N, T, r, S_0, v, q):
    dt = T / N
    brownianPath = np.zeros(N + 1)
    brownianPath[0] = S_0
    for i in range(0, N):
        rand = np.random.normal(0, 1)
        brownianPath[i + 1] = brownianPath[i] * math.e ** ((r - q - .5 * (v ** 2)) * (dt) + v * math.sqrt(dt) * rand)
    return brownianPath

File: FinalCodeC++/brownianMotion.cpp
#include "functions.h"
#include <random>

using namespace std;

double* brownianMotion(int N, double T, double r, double S_0, double v, double q) {
    double dt = T / N;
    static double brownianPath[MAXN];
    
    unsigned seed = rand();
    default_random_engine gen(seed);
    normal_distribution<double> dist(0.0, 1.0);

    brownianPath[0] = S_0;

    for (int i = 0; i < N; ++i) {
        double rand = dist(gen);
        brownianPath[i + 1] = brownianPath[i] * exp((r - q - 0.5 * pow(v, 2)) * dt + v * sqrt(dt) * rand);
    }

    return brownianPath;
}
